error: building geos:x64-windows failed with: BUILD_FAILED

PS D:\work\vcpkg> .\vcpkg.exe install geos
Computing installation plan...
The following packages will be built and installed:
    geos:x64-windows@3.13.0
Detecting compiler hash for triplet x64-windows...
-- Automatically setting %HTTP(S)_PROXY% environment variables to "127.0.0.1:53958".
Compiler found: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe
Restored 0 package(s) from C:\Users\28940\AppData\Local\vcpkg\archives in 119 us. Use --debug to see more details.
Installing 1/1 geos:x64-windows@3.13.0...
Building geos:x64-windows@3.13.0...
-- Using cached geos-3.13.0.tar.bz2.
-- Cleaning sources at D:/work/vcpkg/buildtrees/geos/src/v3.13.0-097c8cf7e3.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/work/vcpkg/downloads/geos-3.13.0.tar.bz2
-- Applying patch fix-exported-config.patch
-- Using source at D:/work/vcpkg/buildtrees/geos/src/v3.13.0-097c8cf7e3.clean
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j33
    Working Directory: D:/work/vcpkg/buildtrees/geos/x64-windows-dbg
    See logs for more information:
      D:\work\vcpkg\buildtrees\geos\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/geos/portfile.cmake:22 (vcpkg_cmake_install)
  scripts/ports.cmake:192 (include)


error: building geos:x64-windows failed with: BUILD_FAILED
See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle geos:x64-windows: 1.1 min
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+geos
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[geos]+Build+error+on+x64-windows&body=Copy+issue+body+from+D%3A%2Fwork%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md

I'm not the original reporter but I have the same problem. The relevant part of the error log is this:

[324/441] C:\PROGRA~2\MICROS~2\2017\COMMUN~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\cl.exe   /TP -DGEOS_DLL_EXPORT -DUSE_UNSTABLE_GEOS_CPP_API -Dgeos_EXPORTS -ID:\vcpkg\buildtrees\geos\src\v3.13.0-097c8cf7e3.clean\include -ID:\vcpkg\buildtrees\geos\x64-windows-dbg\include -ID:\vcpkg\buildtrees\geos\src\v3.13.0-097c8cf7e3.clean\src\deps -ID:\vcpkg\buildtrees\geos\x64-windows-dbg\src\deps /nologo /DWIN32 /D_WINDOWS /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++14 -MDd /fp:precise /showIncludes /FoCMakeFiles\geos.dir\src\operation\relateng\BasicPredicate.cpp.obj /FdCMakeFiles\geos.dir\ /FS -c D:\vcpkg\buildtrees\geos\src\v3.13.0-097c8cf7e3.clean\src\operation\relateng\BasicPredicate.cpp

FAILED: CMakeFiles/geos.dir/src/operation/relateng/BasicPredicate.cpp.obj 

C:\PROGRA~2\MICROS~2\2017\COMMUN~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\cl.exe   /TP -DGEOS_DLL_EXPORT -DUSE_UNSTABLE_GEOS_CPP_API -Dgeos_EXPORTS -ID:\vcpkg\buildtrees\geos\src\v3.13.0-097c8cf7e3.clean\include -ID:\vcpkg\buildtrees\geos\x64-windows-dbg\include -ID:\vcpkg\buildtrees\geos\src\v3.13.0-097c8cf7e3.clean\src\deps -ID:\vcpkg\buildtrees\geos\x64-windows-dbg\src\deps /nologo /DWIN32 /D_WINDOWS /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++14 -MDd /fp:precise /showIncludes /FoCMakeFiles\geos.dir\src\operation\relateng\BasicPredicate.cpp.obj /FdCMakeFiles\geos.dir\ /FS -c D:\vcpkg\buildtrees\geos\src\v3.13.0-097c8cf7e3.clean\src\operation\relateng\BasicPredicate.cpp

D:\vcpkg\buildtrees\geos\src\v3.13.0-097c8cf7e3.clean\include\geos/operation/relateng/TopologyPredicate.h(197): error C2679: binary '<<': no operator found which takes a right-hand operand of type 'std::string' (or there is no acceptable conversion)
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(508): note: could be 'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(std::basic_streambuf<char,std::char_traits<char>> *)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(480): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(const void *)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(460): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(long double)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(440): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(double)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(420): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(float)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(400): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(unsigned __int64)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(380): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(__int64)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(360): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(unsigned long)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(340): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(long)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(320): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(unsigned int)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(295): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(int)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(275): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(unsigned short)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(241): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(short)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(221): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(bool)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(215): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(std::ios_base &(__cdecl *)(std::ios_base &))'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(209): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(std::basic_ios<char,std::char_traits<char>> &(__cdecl *)(std::basic_ios<char,std::char_traits<char>> &))'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(204): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(std::basic_ostream<char,std::char_traits<char>> &(__cdecl *)(std::basic_ostream<char,std::char_traits<char>> &))'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(702): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::operator <<<char,std::char_traits<char>>(std::basic_ostream<char,std::char_traits<char>> &,const char *)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(749): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::operator <<<char,std::char_traits<char>>(std::basic_ostream<char,std::char_traits<char>> &,char)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(787): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::operator <<<std::char_traits<char>>(std::basic_ostream<char,std::char_traits<char>> &,const char *)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(834): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::operator <<<std::char_traits<char>>(std::basic_ostream<char,std::char_traits<char>> &,char)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(960): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::operator <<<std::char_traits<char>>(std::basic_ostream<char,std::char_traits<char>> &,const signed char *)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(967): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::operator <<<std::char_traits<char>>(std::basic_ostream<char,std::char_traits<char>> &,signed char)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(974): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::operator <<<std::char_traits<char>>(std::basic_ostream<char,std::char_traits<char>> &,const unsigned char *)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(981): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::operator <<<std::char_traits<char>>(std::basic_ostream<char,std::char_traits<char>> &,unsigned char)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\ostream(1047): note: or       'std::basic_ostream<char,std::char_traits<char>> &std::operator <<<char,std::char_traits<char>>(std::basic_ostream<char,std::char_traits<char>> &,const std::error_code &)'
D:\vcpkg\buildtrees\geos\src\v3.13.0-097c8cf7e3.clean\include\geos/operation/relateng/TopologyPredicate.h(197): note: while trying to match the argument list '(std::ostream, std::string)'

--- a/ports/geos/portfile.cmake
+++ b/ports/geos/portfile.cmake
@@ -3,11 +3,17 @@ vcpkg_download_distfile(ARCHIVE
     FILENAME "geos-${VERSION}.tar.bz2"
     SHA512 8ffaa3f49a8365db693ac948e9d66cf55321eb12151734c7da2775070b7804ffa607de2474b7019d6ea2a99d5e037fb1e8561bf9025e65ddd4bd1ba049382b28
 )
+vcpkg_download_distfile(msvc_2017_patch
+    URLS https://github.com/libgeos/geos/commit/46e9f158073ebf0d4ec8b7dde37c155d097bc0d7.diff?full_index=1
+    SHA512 9fa1ccc4c66e8268c59bcac218015c2b10ee594bece837e6d0fc78fe700233abd1b2df7aa396c00786ffb170fbfbb0ab530f5007ba10376a2366ee3472d8b02a
+    FILENAME geos-${VERSION}-msvc-2017.diff
+)
 vcpkg_extract_source_archive(SOURCE_PATH
     ARCHIVE "${ARCHIVE}"
     SOURCE_BASE "v${VERSION}"
     PATCHES
         fix-exported-config.patch
+        "${msvc_2017_patch}"
 )
 
 vcpkg_cmake_configure(

--- a/ports/geos/usage
+++ b/ports/geos/usage
@@ -1,10 +1,14 @@
-The package geos provides CMake targets:
+geos provides CMake targets:
 
-    # C API (provides long-term ABI stability)
-    find_package(GEOS CONFIG REQUIRED)
-    target_link_libraries(main PRIVATE GEOS::geos_c)
+  # C API (provides long-term ABI stability)
+  find_package(GEOS CONFIG REQUIRED)
+  target_link_libraries(main PRIVATE GEOS::geos_c)
 
-    # C++ API (will likely change across versions)
-    find_package(GEOS CONFIG REQUIRED)
-    target_link_libraries(main PRIVATE GEOS::geos)
+  # C++ API (will likely change across versions)
+  find_package(GEOS CONFIG REQUIRED)
+  target_link_libraries(main PRIVATE GEOS::geos)
 
+geos provides pkg-config modules:
+
+  # Geometry Engine, Open Source - C API
+  geos

--- a/ports/geos/vcpkg.json
+++ b/ports/geos/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "geos",
   "version": "3.13.0",
+  "port-version": 1,
   "description": "Geometry Engine Open Source",
   "homepage": "https://libgeos.org/",
   "license": "LGPL-2.1-only",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3038,7 +3038,7 @@
     },
     "geos": {
       "baseline": "3.13.0",
-      "port-version": 0
+      "port-version": 1
     },
     "geotrans": {
       "baseline": "3.9",

--- a/versions/g-/geos.json
+++ b/versions/g-/geos.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "beb93ce2a176fe3fc24c5f114dd56c951a60b7b6",
+      "version": "3.13.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "0d124c642d54c1764b3d750e6e8800c0f830c51a",
       "version": "3.13.0",
       "port-version": 0

--- a/versions/g-/geos.json
+++ b/versions/g-/geos.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "beb93ce2a176fe3fc24c5f114dd56c951a60b7b6",
+      "git-tree": "331bb2a4ee2ca09a1d85f801bf3eb52a0ebb2acf",
       "version": "3.13.0",
       "port-version": 1
     },

